pkg/ccl/changefeedccl/alter_changefeed_stmt.go

```diff
@@ -248,6 +248,11 @@ func alterChangefeedPlanHook(
 		}); err != nil {
 			return err
 		}
+		if changefeedbase.FeatureChangefeedDuplicateCheckEnabled.Get(&p.ExecCfg().Settings.SV) {
+			if err := p.ExecCfg().JobRegistry.AddChangefeedIdentityHashToJobInfo(ctx, jobRecord.Details.(jobspb.ChangefeedDetails), job.InfoStorage(p.InternalSQLTxn())); err != nil {
```

Comment:
i'm not 100% sure of the semantics of p.InternalSQLTxn() -- will this be done transactionally with the above update in all cases?